### PR TITLE
[Rust Toolchain] Set rust-version to 1.70.0

### PR DESCRIPTION
**Pull request by bchocho**
### Description

As a follow-up to #8693. To explicitly fail the compilation instead of having potentially confusing compiler mismatch.

https://doc.rust-lang.org/cargo/reference/manifest.html#the-rust-version-field
> If the currently selected version of the Rust compiler is older than the stated version, cargo will exit with an error, telling the user what version is required.

The check found the docker image was old and had to be bumped as well.

### Test Plan

CICD

### CODE REVIEW

**Review by JoshLind — APPROVED** [review]
Thanks @bchocho 😄

**Review by ibalajiarun — APPROVED** [review]

File: Cargo.toml [hunk 1]
```diff
@@ -240,7 +240,7 @@ homepage = "https://aptoslabs.com"
 license = "Apache-2.0"
 publish = false
 repository = "https://github.com/aptos-labs/aptos-core"
-rust-version = "1.64"
+rust-version = "1.70.0"
```
Comment (banool) [inline Cargo.toml:243]: I wonder, do we want to set patch version explicitly, usually it's pretty safe to leave that unspecified right? This way we pick up security fixes for free.

Comment (bchocho) [reply Cargo.toml:243]: This is just a minimum version (according to the doc quoted in the PR description) so it should be fine. IIUC 1.70.0 should be equivalent to 1.70. I'm fine to change it as well if you feel it's different

Comment (bchocho) [reply Cargo.toml:243]: Let me just change it -- the example doesn't have a patch version, so maybe it's not a good idea to include one :)

**Pull request merged**